cobbler/items/item.py

```diff
@@ -134,7 +134,7 @@ def __init__(self, api, is_subobject: bool = False):
         self._template_files = {}
         self._last_cached_mtime = 0
         self._owners: Union[list, str] = enums.VALUE_INHERITED
-        self._cached_dict = ""
+        self._cached_dict: Dict[bool, Any] = {True: None, False: None}
```

Comment:
`{True: None, False: None}` is somewhat magic, can we give it either a descriptive type or at least a comment indicating what the keys are for?

Reply:
@agraul The `to_dict` function has the parameter `resolved: bool = False`, so the cache of this function must also be parameterized and store the result of the execution for each parameter value separately. Fortunately, the boolean type has only 2 values, and we can easily initialize the cache for each `resolved` value, which is the collection key. I'll make a new commit and add a code comment here.

Reply:
@agraul  Added comments to _cached_dict.